### PR TITLE
chore(core): introduce mixed I/O config method

### PR DESCRIPTION
**Pull request by mariusgheorghies**
Introduces a `CairoConfiguration#isWriterMixedIOEnabled()` method to be used in `TableWriter` to enable/disable mixed mmap and `pwrite` I/O. Also, cleans up tear-down methods across `AbstractCairoTest` and `AbstractGriffinTest` implementations.

Also, ports required `Bootstrap`-related changes from https://github.com/questdb/questdb/tree/vi_2

### CODE REVIEW

File: core/README.md [hunk 1]
```diff
@@ -244,3 +244,4 @@ You can mount host directories using `-v` option, e.g.
 For more details, see the
 [official Docker documentation](https://questdb.io/docs/get-started/docker) for
 working with QuestDB.
+//
```
Comment (bluestreak01) [inline core/README.md:247]: @puzpuzpuz may be this is leftover

**Comment by bluestreak01** [comment]
also ZFS build does not run tests. May this is why it is so green? :)

image

**Comment by mariusgheorghies** [comment]
> also ZFS build does not run tests. May this is why it is so green? :)

That's because i removed the // from core/README.md

**Comment by mariusgheorghies** [comment]
> @mariusgheorghies why do we need this:
> 
> image

we do not as the azure hosted agents run on ext4

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 4 (75.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 30 (96.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 30 (96.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 30 (96.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 30 (96.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 30 (96.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 30 (96.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 30 (96.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 30 (96.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 30 (96.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 30 (96.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 30 (96.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 30 (96.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 30 (96.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 30 (96.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Files.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 53 / 56 (94.64%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/Bootstrap.java|25|26|96.15%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/DefaultBootstrapConfiguration.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 52 / 56 (92.86%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/FilesFacadeImpl.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/Bootstrap.java|23|24|95.83%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/DefaultBootstrapConfiguration.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 52 / 56 (92.86%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/FilesFacadeImpl.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/Bootstrap.java|23|24|95.83%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/DefaultBootstrapConfiguration.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 52 / 56 (92.86%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/FilesFacadeImpl.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/Bootstrap.java|23|24|95.83%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/DefaultBootstrapConfiguration.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

**Comment by puzpuzpuz** [comment]
Closing in favor of #3145

**Pull request closed without merging**